usng4j-impl/src/main/java/org/codice/usng4j/impl/CoordinateSystemTranslatorImpl.java

```diff
@@ -23,27 +23,36 @@
 
 package org.codice.usng4j.impl;
 
+import static org.codice.usng4j.NSIndicator.NORTH;
+import static org.codice.usng4j.NSIndicator.SOUTH;
+
 import java.text.ParseException;
 import java.util.Optional;
 import org.codice.usng4j.BoundingBox;
 import org.codice.usng4j.CoordinatePrecision;
 import org.codice.usng4j.CoordinateSystemTranslator;
 import org.codice.usng4j.DecimalDegreesCoordinate;
-import org.codice.usng4j.NSIndicator;
+import org.codice.usng4j.UpsCoordinate;
 import org.codice.usng4j.UsngCoordinate;
 import org.codice.usng4j.UtmCoordinate;
+import org.codice.usng4j.UtmUpsCoordinate;
 
-/** {@inheritDoc} */
 public final class CoordinateSystemTranslatorImpl implements CoordinateSystemTranslator {
 
   public static final double NORTHING_OFFSET = 10000000.0; // (meters)
 
-  private static final double DEG_2_RAD = Math.PI / 180;
+  private static final double DEG_2_RAD = Math.PI / 180.0;
 
   private static final double RAD_2_DEG = 180.0 / Math.PI;
 
   private static final int BLOCK_SIZE = 100000;
 
+  private static final double EPSILON = Math.ulp(1.0);
```

Comment:
❓ Why is this the correct value for `EPSILON` not not `Math.ulp(2.0)` or any other input float? Is it somewhat arbitrary, determined empirically, or dictated by some aspect of the spec?

Reply:
This epsilon value is intended to be the smallest representable difference between 1.0 and the next double value to match the behavior of these calculations from the original C++ code that it was ported from (which uses [numeric_limits::epsilon](https://en.cppreference.com/w/cpp/types/numeric_limits/epsilon)) and from usng.js (which uses [Number.EPSILON](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/EPSILON)).